tests/test-class-amp-theme-support.php

```diff
@@ -67,4 +67,53 @@ public function test_is_paired_available() {
 		$this->assertTrue( is_search() );
 		$this->assertFalse( AMP_Theme_Support::is_paired_available() );
 	}
+
+	/**
+	 * Test finish_output_buffering.
+	 *
+	 * @covers AMP_Theme_Support::finish_output_buffering()
+	 */
+	public function test_finish_output_buffering() {
+		add_theme_support( 'amp' );
+		AMP_Theme_Support::init();
+		ob_start();
+		?>
+		<!DOCTYPE html>
+		<html amp <?php language_attributes(); ?>>
+			<head>
+				<?php wp_head(); ?>
+				<script data-head>document.write('TODO: This needs to be sanitized as well once.');</script>
+			</head>
+			<body>
+				<img width="100" height="100" src="https://example.com/test.png">
+				<audio width="400" height="300" src="https://example.com/audios/myaudio.mp3"></audio>
+				<amp-ad type="a9"
+					width="300"
+					height="250"
+					data-aax_size="300x250"
+					data-aax_pubname="test123"
+					data-aax_src="302"></amp-ad>
+				<?php wp_footer(); ?>
+			</body>
+		</html>
+		<?php
+		$original_html  = trim( ob_get_clean() );
+		$sanitized_html = AMP_Theme_Support::finish_output_buffering( $original_html );
+
+		$this->assertContains( '<meta charset="utf-8">', $sanitized_html );
```

Comment:
@westonruter @DavidCramer these tests could have a bit more coverage here. I believe it is ok to do that at a later stage since the other components sanitization (form etc.) are still WIP.